pandas/core/generic.py

```diff
@@ -7845,16 +7845,21 @@ def interpolate(
             obj, should_transpose = self, False
         else:
             obj, should_transpose = (self.T, True) if axis == 1 else (self, False)
-            if np.any(obj.dtypes == object):
-                # GH#53631
-                if not (obj.ndim == 2 and np.all(obj.dtypes == object)):
-                    # don't warn in cases that already raise
-                    warnings.warn(
-                        f"{type(self).__name__}.interpolate with object dtype is "
-                        "deprecated and will raise in a future version. Call "
-                        "obj.infer_objects(copy=False) before interpolating instead.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
+            # GH#53631
+            if obj.ndim == 1 and obj.dtype == object:
+                raise TypeError(
+                    f"{type(self).__name__} cannot interpolate with object dtype."
+                )
+            if obj.ndim == 2:
+                if np.all(obj.dtypes == object):
+                    raise TypeError(
+                        "Cannot interpolate with all object-dtype columns "
+                        "in the DataFrame. Try setting at least one "
+                        "column to a numeric dtype."
+                    )
```

Comment:
I think we can now simplify things and remove this check. The reason: we raise
```
TypeError(
    f"{type(self).__name__} cannot interpolate with object dtype."
)
```

in case `np.any(obj.dtypes == object)` below and there is no need to raise Type Error `if np.all(obj.dtypes == object)`. Should I remove raising
```
if np.all(obj.dtypes == object):
    raise TypeError(
        "Cannot interpolate with all object-dtype columns "
        "in the DataFrame. Try setting at least one "
        "column to a numeric dtype."
    )
```
?

Reply:
Nice catch! Yeah try that out

Reply:
But the messaging will need to probably change, this could be a `Series`

Reply:
sorry, I don't understand why is it a `Series`? I thought it's a `DataFrame`, because we are in the block
`if obj.ndim == 2:`

I suggest to use the message  `f"{type(self).__name__} cannot interpolate with object dtype."`, the same one we use in the block below
```
elif np.any(obj.dtypes == object):
    raise TypeError(
        f"{type(self).__name__} cannot interpolate with object dtype."
    )
```
I think we can just remove the block with the old message:
```
if obj.ndim == 2:
    if np.all(obj.dtypes == object):
...
```
and change the old message in tests to the new one `"DataFrame cannot interpolate with object dtype"`

Reply:
Ah sorry I had misunderstood your initial suggestion.

You can also go a step further and replace All of these checks with

```python
if np.any(obj.dtypes == object):
    raise TypeError(f"{type(self).__name__} cannot interpolate with object dtype.")
```

Since `Series.dtypes` and `DataFrame.dtypes` can be checked at the same time

Reply:
sorry, my wording wasn't clear. 

My idea was to remove the second part of the error message:
```
"Cannot interpolate with all object-dtype columns in the DataFrame. Try setting at least one column to a numeric dtype."
```
`"Try setting at least one column to a numeric dtype"` won't work from now on because of the enforcing in this PR. Then I thought why not to replace the first part: `"Cannot interpolate with all object-dtype columns in the DataFrame."` with a new msg: `"... cannot interpolate with object dtype."`
If we do this we can remove one check block.

I made changes. Could you please take a look at it?